crates/walrus-service/src/client/cli/args.rs

```diff
@@ -1011,6 +1138,139 @@ impl std::fmt::Display for BlobIdentity {
     }
 }
 
+/// Query for quilt patches by identifier.
+#[derive(Debug, Clone, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct QuiltPatchByIdentifier {
+    /// The quilt ID, which is the BlobID of the quilt.
+    pub quilt_id: BlobId,
+    /// The identifiers to read from the quilt.
+    pub identifiers: Vec<String>,
+}
+
+/// Query for quilt patches by tag.
+#[derive(Debug, Clone, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct QuiltPatchByTag {
+    /// The quilt ID, which is the BlobID of the quilt.
+    pub quilt_id: BlobId,
+    /// The tag key.
+    pub tag: String,
+    /// The tag value.
+    pub value: String,
+}
+
+/// Query for quilt patches by patch ID.
+#[derive(Debug, Clone, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct QuiltPatchByPatchId {
+    /// The quilt patch IDs.
+    pub quilt_patch_ids: Vec<QuiltPatchId>,
+}
+
+/// Selector for quilt patches.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum QuiltPatchSelector {
+    /// Patches by quilt_id and identifier.
+    ByIdentifier(QuiltPatchByIdentifier),
+    /// Patches by quilt_id and tag.
+    ByTag(QuiltPatchByTag),
+    /// Patches by quilt_patch_id.
+    ByPatchId(QuiltPatchByPatchId),
+}
+
+/// Query parameters to read quilt patches.
+#[serde_as]
+#[derive(Debug, Clone, Args, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct QuiltPatchQuery {
+    /// The quilt ID, which is the BlobID of the quilt.
+    #[serde_as(as = "Option<DisplayFromStr>")]
+    #[arg(long, allow_hyphen_values = true, value_parser = parse_blob_id)]
+    pub quilt_id: Option<BlobId>,
+
+    /// The identifiers to read from the quilt.
+    #[arg(long = "identifier", action = clap::ArgAction::Append)]
+    #[serde(default)]
+    pub identifiers: Vec<String>,
+
+    /// The tag key.
+    #[arg(long)]
+    pub tag: Option<String>,
+
+    /// The tag value.
+    #[arg(long)]
+    pub value: Option<String>,
```

Comment:
Question: Could we allow specifying multiple tags? (Can also be in a follow-up.)